L1Trigger/L1TTrackMatch/plugins/L1TkMuonProducer.cc

```diff
@@ -0,0 +1,712 @@
+    dwcorr_->set_min_trk_p(iConfig.getParameter<double>("min_trk_p"));
+    dwcorr_->set_max_trk_aeta(iConfig.getParameter<double>("max_trk_aeta"));
+    dwcorr_->set_max_trk_chi2(iConfig.getParameter<double>("max_trk_chi2"));
+    dwcorr_->set_min_trk_nstubs(iConfig.getParameter<int>("min_trk_nstubs"));
+    dwcorr_->set_do_trk_qual_presel(true);
+  }
+
+  if (bmtfMatchAlgoVersion_ == kMantra) {
+    std::string fIn_bounds_name = iConfig.getParameter<edm::FileInPath>("mantra_bmtfcorr_boundaries").fullPath();
+    std::string fIn_theta_name = iConfig.getParameter<edm::FileInPath>("mantra_bmtfcorr_theta_windows").fullPath();
+    std::string fIn_phi_name = iConfig.getParameter<edm::FileInPath>("mantra_bmtfcorr_phi_windows").fullPath();
+
+    auto bounds = L1TkMuMantra::prepare_corr_bounds(fIn_bounds_name, "h_dphi_l");
+
+    TFile* fIn_theta = TFile::Open(fIn_theta_name.c_str());
+    TFile* fIn_phi = TFile::Open(fIn_phi_name.c_str());
+
+    mantracorr_barr_ = std::unique_ptr<L1TkMuMantra>(new L1TkMuMantra(bounds, fIn_theta, fIn_phi, "mantra_barrel"));
+
+    fIn_theta->Close();
+    fIn_phi->Close();
+
+    // settings : NB : now hardcoded, could be read from cfg
+    mantracorr_barr_->set_safety_factor(0.5, 0.5);
+    mantracorr_barr_->setArbitrationType("MaxPt");
+  }
+
+  if (omtfMatchAlgoVersion_ == kMantra) {
+    std::string fIn_bounds_name = iConfig.getParameter<edm::FileInPath>("mantra_omtfcorr_boundaries").fullPath();
+    std::string fIn_theta_name = iConfig.getParameter<edm::FileInPath>("mantra_omtfcorr_theta_windows").fullPath();
+    std::string fIn_phi_name = iConfig.getParameter<edm::FileInPath>("mantra_omtfcorr_phi_windows").fullPath();
+
+    auto bounds = L1TkMuMantra::prepare_corr_bounds(fIn_bounds_name, "h_dphi_l");
+
+    TFile* fIn_theta = TFile::Open(fIn_theta_name.c_str());
+    TFile* fIn_phi = TFile::Open(fIn_phi_name.c_str());
+
+    mantracorr_ovrl_ = std::unique_ptr<L1TkMuMantra>(new L1TkMuMantra(bounds, fIn_theta, fIn_phi, "mantra_overlap"));
+
+    fIn_theta->Close();
+    fIn_phi->Close();
+
+    // settings : NB : now hardcoded, could be read from cfg
+    mantracorr_ovrl_->set_safety_factor(0.5, 0.5);
+    mantracorr_ovrl_->setArbitrationType("MaxPt");
+  }
+
+  if (emtfMatchAlgoVersion_ == kMantra) {
+    std::string fIn_bounds_name = iConfig.getParameter<edm::FileInPath>("mantra_emtfcorr_boundaries").fullPath();
+    std::string fIn_theta_name = iConfig.getParameter<edm::FileInPath>("mantra_emtfcorr_theta_windows").fullPath();
+    std::string fIn_phi_name = iConfig.getParameter<edm::FileInPath>("mantra_emtfcorr_phi_windows").fullPath();
+
+    auto bounds = L1TkMuMantra::prepare_corr_bounds(fIn_bounds_name, "h_dphi_l");
+
+    TFile* fIn_theta = TFile::Open(fIn_theta_name.c_str());
+    TFile* fIn_phi = TFile::Open(fIn_phi_name.c_str());
+
+    mantracorr_endc_ = std::unique_ptr<L1TkMuMantra>(new L1TkMuMantra(bounds, fIn_theta, fIn_phi, "mantra_endcap"));
```

Comment:
use make_unique (if this needs to be a pointer at all)

Reply:
done